fre/app/regrid_xy/regrid_xy_rewrite.py

```diff
@@ -0,0 +1,257 @@
+  """
+  
+  if grid_spec is None: grid_spec = "grid_spec.nc"
+    
+  #retrieve the grid_spec tar filename from ppyaml
+  pp_grid_spec_tar = datadict["pp_yaml"]['postprocess']['settings']['pp_grid_spec']
+    
+  #untar grid_spec tar file
+  if tarfile.is_tarfile(pp_grid_spec_tar):
+    with tarfile.open(pp_grid_spec_tar, 'r') as tar: tar.extractall() 
+        
+  if Path(grid_spec).exists():
+    fre_logger(f"Grid_spec file {grid_spec} found in {pp_grid_spec_tar}")
+    return grid_spec
+  else:
+    raise IOError(f"Grid_spec={grid_spec} could not be found in {pp_grid_spec_tar}")
+    
+
+def get_input_file(datadict: dict, history_file: str):
+  
+  """
+  Gets the 
+  """
+  
+  input_date = datadict["input_date"]
+  if input_date is None:
+    return history_file
+  else:
+    return f"{input_date}.{history_file}"
+
+
+def get_remap_file(datadict: dict):
+
+  remap_file = datadict["remap_file"]
+  input_dir = datadict["input_dir"]
+  input_mosaic = datadict["input_mosaic"]
+  nlon = datadict["output_nlon"]
+  nlat = datadict["output_nlat"]
+  interp_method = datadict["interp_method"]  
+    
+  remap_cache_file = f"{input_mosaic[:-3]}X{nlon}by{nlat}_{interp_method}.nc"
+  if remap_file is not None:
+    try:
+      shutil.copy(remap_file, input_dir+remap_cache_file)
+      fre_logger.info(f"Copied/Renamed {remap_file} to {remap_cache_file}")
+    except Exception as exc:
+      fre_logger.info(f"Cannot find specified remap_file {remap_file}\n"
+                      "Remap file {remap_cache_file} will be generated on-the-fly")
+
+  return remap_cache_file
+
+
+def get_scalar_fields(datadict: dict):
+
+    """
+    Returns all the regriddable variables in an input file
+    """
+    
+    mosaic_file = datadict["input_mosaic"]
+    input_file = datadict["input_file"]
+    
+    if xr.load_dataset(str(mosaic_file)).sizes["ntiles"] > 1:
+      input_file += ".tile1.nc"
+    else:
+      input_file += ".nc"
+    
+    #load data file
+    #xarray gives an error if variables to drop do not exist in the file.  The errors="ignore" fixes that
+    dataset = xr.load_dataset(str(input_file)).drop_vars(non_regriddable_variables, errors="ignore")
+
+    #list of variables to regrid, only multi-dimensional data will be regridded
+    regrid_vars_string = ",".join([variable for variable in dataset if len(dataset[variable].sizes)>1])
+
+    return regrid_vars_string
+
+
+def write_summary(datadict):
+  
+  fre_logger("JOB SUMMARY")
+  fre_logger(f"FREGRID input directory: {datadict['input_dir']}")
+  fre_logger(f"FREGRID output_directory: {datadict['output_dir']}")
+  fre_logger(f"FREGRID input mosaic file: {datadict['input_mosaic']}")
+  fre_logger(f"FREGRID output lonxlat grid: {datadict['output_nlon']} X {datadict['output_nlat']}")
+  fre_logger(f"FREGRID remap file: {datadict['remap_file']}")
+  
+
+def regrid_xy(pp_yamlfile: str,
+              input_dir: str,
+              output_dir: str,
+              remap_file: str = None,
```

Comment:
Let's omit this.

The remap file will be created in a canonical name with uniqueness of (input resolution, output resolution, and interp method), and stashing in the inputdir is fine.

Reply:
Done